riptable/rt_matplotlib.py

```diff
@@ -0,0 +1,754 @@
+from typing import List, Optional
+import datetime
+import pytz
+import dateutil.rrule
+from matplotlib import units, ticker, dates
+
+import riptable as rt
+
+sizes_ns = {
+    'MICROSECONDLY': 1_000,
+    'SECONDLY': 1_000_000_000,
+    'MINUTELY': 60 * 1_000_000_000,
+    'HOURLY': 60 * 60 * 1_000_000_000,
+    'DAILY': 24 * 60 * 60 * 1_000_000_000,
+    'MONTHLY': 30 * 24 * 60 * 60 * 1_000_000_000,
+    'YEARLY': 365 * 24 * 60 * 60 * 1_000_000_000,
+}
+
+NANOS_PER_SECOND = 1_000_000_000
+MICROS_PER_SECOND = 1_000_000
+
+
+def dts_to_ns(dts):
+    return [dt_to_ns(dt) for dt in dts]
+
+
+def td_to_ns(td):
+    ns = td.days * sizes_ns['DAILY'] + td.seconds * sizes_ns['SECONDLY'] + td.microseconds * 1_000
+    return ns
+
+
+def dt_to_ns(dt):
+    return int(dt.timestamp() * NANOS_PER_SECOND)
+
+
+class DTTicker:
```

Comment:
It looks like ``DTTicker`` is effectively an _abstract_ base class (i.e. that we won't ever instantiate directly) -- is that correct? If so, consider enforcing that by having it derive from ``abc.ABC``, then decorating the abstract methods (the ones that need to be implemented by the derived classes) with ``@abc.abstractmethod``.

Docs for the ``abc`` module: https://docs.python.org/3/library/abc.html